src/java/org/apache/cassandra/cql3/CqlConstraintFunctionExecutor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3;
+
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.cassandra.schema.TableMetadata;
+
+/**
+ * Interface to be implemented by functions that are executed as part of CQL constraints.
+ */
+public interface CqlConstraintFunctionExecutor
```

Comment:
Hmmm... the name "executor" is confusing. I thought it was an executor for concurrency. But it is not according to the implementations. Can you rename the interface?  Or can it be deleted and just keep `ConstraintFunction`?